### ISSUE TITLE
Documentation: Missing SYNOPSIS in App::Cmd::Subdispatch (rt64299)

### ISSUE THREAD

**Issue by rjbs** [rjbs/App-Cmd#17]
https://rt.cpan.org/Ticket/Display.html?id=64299

I know in principle what A::C::S is supposed to do, but I have no idea
how to use it in my code. Even just a single example would help...
## Mark.

Mark Lawrence

**Comment by jayallen** [comment]
Yeah, I actually pored through the tests to find out and—after a massive number of issues caused, I think, primarily because I was also using a base command class in addition to a nested command—I finally did...but then I ran into #35 and gave up.

It would be great to see:

1. Better docs around this including examples of using a base command class in addition to a subcommand
3. Support in App::Cmd::Setup for subdispatch to keep things consistent
4. Fixing #35 because the current display of subcommands is too confusing to unleash on users
5. Examples of using subcommands with the same names (e.g. `myapp foo list` and `mysql bar list`) which gave me errors because of duplicate command names (though I admit without good documentation I may have been doing it wrong)
